### PR TITLE
[#173154240] callActivityWithRetry for all Activity invocations

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: DeleteEligibilityCheckActivity/handler.ts [hunk 1]
```diff
@@ -63,7 +63,15 @@ export const getDeleteEligibilityCheckActivityHandler = (
               if (err.code === 404) {
                 return fromEither(right("NOT FOUND"));
               }
-              return fromEither(left(new Error(`QueryError: [${err}]`)));
+              // this condition address the case we want the activity to throw, so the orchestrator can retry
+              const queryError = new Error(
```
Comment (gunzip) [inline DeleteEligibilityCheckActivity/handler.ts:67]: can you please fix in the code above the double mapping / casting from fiscalCode to NonEmptyString?

File: FailedBonusActivationActivity/__tests__/handler.test.ts [hunk 2]
```diff
@@ -122,9 +122,14 @@ describe("FailedBonusActivationHandler", () => {
...
-    const response = await handler(context, {
-      bonusActivation: aBonusActivationWithFamilyUID
-    });
-    expect(UnhandledFailure.decode(response).isRight()).toBeTruthy();
+    try {
+      await handler(context, {
+        bonusActivation: aBonusActivationWithFamilyUID
+      });
+      // expect that the activity fails
+      expect(false).toBeTruthy();
```
Comment (gunzip) [inline FailedBonusActivationActivity/__tests__/handler.test.ts:130]: you can use `fail()`

File: StartBonusActivationOrchestrator/handler.ts [hunk 3]
```diff
@@ -82,8 +83,10 @@ export const getStartBonusActivationOrchestratorHandler = (
         name: "bonus.activation.success"
       });
     } else {
-      yield context.df.callActivity(
+      // TODO: is FailedBonusActivationActivity idempotent?
```
Comment (gunzip) [inline StartBonusActivationOrchestrator/handler.ts:86]: cc @balanza

File: SuccessBonusActivationActivity/handler.ts [hunk 4]
```diff
@@ -41,13 +41,8 @@ export const UnhandledFailure = t.interface({
...
 export type UnhandledFailure = t.TypeOf<typeof UnhandledFailure>;
 
-export const TransientFailure = t.interface({
-  kind: t.literal("TRANSIENT")
-});
-export type TransientFailure = t.TypeOf<typeof TransientFailure>;
-
 const SuccessBonusActivationFailure = t.union(
-  [InvalidInputFailure, UnhandledFailure, TransientFailure],
+  [InvalidInputFailure, UnhandledFailure],
```
Comment (gunzip) [inline SuccessBonusActivationActivity/handler.ts:45]: why this change here?

Comment (BurnedMarshal) [reply SuccessBonusActivationActivity/handler.ts:45]: `TransientFailure` never happens inside this activity, all errors are remapped to `UnhandledFailure`. The Input decode error is instead `InvalidInputFailure`

File: FailedBonusActivationActivity/__tests__/handler.test.ts (hunk 2 above)
Comment (BurnedMarshal) [reply FailedBonusActivationActivity/__tests__/handler.test.ts]: Fixed in 9011499

File: DeleteEligibilityCheckActivity/handler.ts (hunk 1 above)
Comment (BurnedMarshal) [reply DeleteEligibilityCheckActivity/handler.ts:62]: Do you mean this 87fb0ed?

Comment (gunzip) [reply DeleteEligibilityCheckActivity/handler.ts:62]: yes :)

File: FailedBonusActivationActivity/handler.ts [hunk 5]
```diff
@@ -173,6 +173,13 @@ export function FailedBonusActivationHandler(
         l => l,
         () => FailedBonusActivationSuccess.encode({ kind: "SUCCESS" })
       )
+      .map(result => {
+        // this condition address the case we want the activity to throw, so the orchestrator can retry
+        if (UnhandledFailure.decode(result).isRight()) {
```
Comment (gunzip) [inline FailedBonusActivationActivity/handler.ts:178]: Since this error is threated as transient, can we rename it to `TransientFailure`?

File: StartBonusActivationOrchestrator/handler.ts (hunk 3 above)
Comment (gunzip) [reply StartBonusActivationOrchestrator/handler.ts:86]: what's needed to state this?

File: SuccessBonusActivationActivity/handler.ts (hunk 4 above)
Comment (gunzip) [reply SuccessBonusActivationActivity/handler.ts:45]: if an error cause a retry it's better to call it `TransientFailure` so we should rename this (as it's handled actually : )

Comment (BurnedMarshal) [reply SuccessBonusActivationActivity/handler.ts]: Fixed in 790c445

File: FailedBonusActivationActivity/handler.ts (hunk 5 above)
Comment (BurnedMarshal) [reply FailedBonusActivationActivity/handler.ts]: Fixed in 790c445

File: StartBonusActivationOrchestrator/handler.ts (hunk 3 above)
Comment (BurnedMarshal) [reply StartBonusActivationOrchestrator/handler.ts:86]: `FailedBonusActivationActivity` use queries for update/delete data that don't throw errors. Only relaseLockForUserFamily does. Do we want a retry even on other queries failures?

Comment (gunzip) [reply StartBonusActivationOrchestrator/handler.ts]: yes, it looks to me it's idempotent. what do you think about it?

File: FailedBonusActivationActivity/handler.ts [hunk 6]
```diff
@@ -163,16 +153,22 @@ export function FailedBonusActivationHandler(
           context.log.warn(
             `FailedBonusActivationHandler|WARN|Failed releasing lock: ${err.body}`
```
Comment (gunzip) [inline FailedBonusActivationActivity/handler.ts:154]: @balanza why it's a warn and not an error here ? are there scenarios where failing to release the lock it's ok? (ie. the lock wasn't created at all and we're trying here to delete something that does not exists?)

File: SuccessBonusActivationActivity/handler.ts [hunk 7]
```diff
@@ -131,9 +121,8 @@ export function SuccessBonusActivationHandler(
             context.log.warn(
               `FailedBonusActivationHandler|WARN|Failed updating bonus: ${err.body}`
```
Comment (gunzip) [inline SuccessBonusActivationActivity/handler.ts:122]: same here, why a warn and not an error?
